### PR TITLE
Task reassigned to reviewer when resubmitted

### LINKED ISSUE #632: reassign task to reviewer when resubmitted after changes requested
add a `reviewed_by_user` to the Task model, set that to whoever `unsubmit`s a task, and assign it to them when the task is resubmitted (we can check whenever a task is submitted whether it's got `changes_requested`, and send an email to the `reviewed_by_user` saying something like `User Y has resubmitted the following task to which you requested changes: …`?

### PR DESCRIPTION
**Pull request by goose-life**
closes #632

### CODE REVIEW

File: indigo_api/models.py [hunk 1]
```diff
@@ -1263,8 +1263,8 @@ def may_submit(self, view):
     def submit(self, user):
         if not self.assigned_to:
             self.assign_to(user, user)
-        self.last_assigned_to = self.assigned_to
-        self.assigned_to = None
+        self.submitted_by_user = self.assigned_to
+        self.assigned_to = self.reviewed_by_user
```
Comment (longhotsummer) [inline indigo_api/models.py]: Shouldn't this be `self.assign_to` so that the assignment is stored as an action and emails are sent?

Comment (longhotsummer) [reply indigo_api/models.py]: Oh, I see that there's a new email for `submit` with a reviewer. I'm not against having  both emails sent: assigned to you, and the task was submitted back to you for review.

Comment (goose-life) [reply indigo_api/models.py]: I'm not crazy about the idea of two emails – I think of it as the converse of 'changes requested', where only one email is sent to the submitted_by_user. Happy to use the assignment instead, the wording of the email was just a bit trickier (I tried that first).

Comment (longhotsummer) [reply indigo_api/models.py]: Fair point, it mirrors `unsubmit`. Let's do it.

**Pull request merged**